File: medium/692-top-k-frequent-words/692-top-k-frequent-words.rb
# @param {String[]} words
# @param {Integer} k
# @return {String[]}
def top_k_frequent(words, k)
    freqs = words.tally.entries.sort { |a, b| b[1] != a[1] ? b[1] <=> a[1] : a[0] <=> b[0] }
    freqs[0...k].map { |e| e[0] }
end

File: medium/692-top-k-frequent-words/692-top-k-frequent-words.js
/**
 * @param {string[]} words
 * @param {number} k
 * @return {string[]}
 */
const topKFrequent = function(words, k) {
    const freqMap = {};
    words.forEach(word => freqMap[word] = ++freqMap[word] || 1);
    return Object.keys(freqMap)
            .sort((a, b) => {
                if (freqMap[a] === freqMap[b]) {
                    return b < a ? 1 : -1 
                }
                return freqMap[b] - freqMap[a];
            })
            .slice(0, k)
};